src/backend/tests/teams.test.ts

```diff
@@ -326,4 +326,50 @@ describe('Teams', () => {
       expect(res).toStrictEqual(sharedTeam1);
     });
   });
+
+  describe('Archive team', () => {
+    test('Archive team doesn`t work if the team is not found', async () => {
+      await expect(async () => await TeamsService.archiveTeam(batman, sharedTeam1.teamId)).rejects.toThrow(
+        new NotFoundException('Team', sharedTeam1.teamId)
+      );
+    });
+
+    test('Archive team doesn`t work if the user is not an admin or above', async () => {
+      vi.spyOn(prisma.team, 'findFirst').mockResolvedValue(justiceLeague);
+      await expect(async () => await TeamsService.archiveTeam(theVisitor, primsaTeam2.teamId)).rejects.toThrow(
+        new AccessDeniedException('You must be an admin or above to archive a team')
+      );
+    });
+
+    test('Archive team doesn`t work if a project in the team is incomplete or active', async () => {
+      vi.spyOn(prisma.team, 'findFirst').mockResolvedValue(primsaTeam2);
+      await expect(async () => await TeamsService.archiveTeam(superman, primsaTeam2.teamId)).rejects.toThrow(
+        new HttpException(400, 'A team is not archivable if it has any active projects, or incomplete projects')
+      );
+    });
+
+    test('Archive team works', async () => {
+      vi.spyOn(prisma.team, 'findFirst').mockResolvedValue(justiceLeague);
+
+      const justiceLeagueUpdates = { ...justiceLeague, userArchivedId: 2, dateArchived: new Date() };
+
+      vi.spyOn(prisma.team, 'update').mockResolvedValue(justiceLeagueUpdates);
+      const res = await TeamsService.archiveTeam(superman, justiceLeague.teamId);
+      const { teamId } = justiceLeague;
+
+      expect(prisma.team.findFirst).toBeCalledTimes(1);
+      expect(prisma.team.update).toBeCalledTimes(1);
+      expect(prisma.team.update).toHaveBeenCalledWith({
+        where: { teamId },
+        ...teamQueryArgs,
+        data: {
+          userArchivedId: superman.userId,
+          dateArchived: justiceLeague.dateArchived !== null ? new Date() : null
+        }
+      });
+
+      expect(res.userArchivedId).toBe(justiceLeagueUpdates.userArchivedId);
```

Comment:
the mock call above passes, so you don't need to check the response in line 371/2. It's not really testing anything more, so it's not worth trying to mess with the janky mock stuff